packages/mdc-select/foundation.js

```diff
@@ -77,47 +84,89 @@ class MDCSelectFoundation extends MDCFoundation {
     super(Object.assign(MDCSelectFoundation.defaultAdapter, adapter));
   }
 
-  /**
-   * Updates the styles of the select to show the disasbled state.
-   * @param {boolean} disabled
-   */
-  updateDisabledStyle(disabled) {
-    const {DISABLED} = MDCSelectFoundation.cssClasses;
-    if (disabled) {
-      this.adapter_.addClass(DISABLED);
-    } else {
-      this.adapter_.removeClass(DISABLED);
-    }
+  setSelectedIndex(index) {
+    this.adapter_.setSelectedIndex(index);
+    this.adapter_.closeMenu();
+    const didChange = true;
+    this.handleChange(didChange);
+  }
+
+  setValue(value) {
+    this.adapter_.setValue(value);
+    const didChange = true;
+    this.handleChange(didChange);
+  }
+
+  getValue() {
+    return this.adapter_.getValue();
+  }
+
+  setDisabled(isDisabled) {
```

Comment:
I noticed you removed the code that adds/removes a CSS class in `setDisabled`... isn't that still needed?

It looks like you baked in the root class addition/removal into the adapter for enhanced only, but AFAICT it is still needed for both variants, so I'm not sure why that was removed from the foundation.